connect/runtime/src/main/java/org/apache/kafka/connect/util/LoggingContext.java

```diff
@@ -0,0 +1,121 @@
+         */
+        public static final String CONNECTOR_NAME = "connector.name";
+
+        /**
+         * The name of the Mapped Diagnostic Context (MDC) key that defines the task number of a <i>connector</i>.
+         */
+        public static final String CONNECTOR_TASK = "connector.task";
+
+        /**
+         * The name of the Mapped Diagnostic Context (MDC) key that defines a <i>scope</i> within a connector.
+         */
+        public static final String CONNECTOR_SCOPE = "connector.scope";
+    }
+
+    /**
+     * The {@link Parameters#CONNECTOR_SCOPE scope} values used by Connect.
+     */
+    public enum Scope {
+        /**
+         * The {@link Parameters#CONNECTOR_SCOPE scope} value for the worker as it starts a connector.
+         */
+        WORKER("worker"),
+
+        /**
+         * The {@link Parameters#CONNECTOR_SCOPE scope} value for {@link org.apache.kafka.connect.connector.Task}
+         * implementations.
+         */
+        TASK("task"),
+
+        /**
+         * The {@link Parameters#CONNECTOR_SCOPE scope} value for committing offsets.
+         */
+        OFFSETS("offsets");
+
+        private final String value;
+        Scope(String value) {
+            this.value = value;
+        }
+        public String value() {
+            return this.value;
+        }
+    }
+
+    /**
+     * Modify the current {@link MDC} logging context to set the {@link Parameters#CONNECTOR_NAME connector name} to the
+     * supplied name and the {@link Parameters#CONNECTOR_SCOPE scope} to {@link Scope#WORKER}.
+     *
+     * @param connectorName the connector name; may not be null
+     */
+    public static void forConnector(String connectorName) {
+        MDC.put(Parameters.CONNECTOR_NAME, connectorName);
+        MDC.remove(Parameters.CONNECTOR_TASK);
+        MDC.put(Parameters.CONNECTOR_SCOPE, Scope.WORKER.value());
+    }
+
+    /**
+     * Modify the current {@link MDC} logging context to set the {@link Parameters#CONNECTOR_NAME connector name} and
+     * {@link Parameters#CONNECTOR_TASK task number} to the same values in the supplied {@link ConnectorTaskId}.
+     *
+     * @param id the connector task ID; may not be null
+     */
+    public static void forTask(ConnectorTaskId id) {
+        MDC.put(Parameters.CONNECTOR_NAME, id.connector());
+        MDC.put(Parameters.CONNECTOR_TASK, Integer.toString(id.task()));
+        MDC.put(Parameters.CONNECTOR_SCOPE, Scope.TASK.value());
+    }
+
+    /**
+     * Modify the current {@link MDC} logging context to set the  {@link Parameters#CONNECTOR_NAME connector name},
+     * {@link Parameters#CONNECTOR_TASK task number} and to the same values in the supplied {@link ConnectorTaskId}, and to set
+     * the scope to {@link Scope#OFFSETS}.
+     *
+     * @param id the connector task ID; may not be null
+     */
+    public static void forOffsets(ConnectorTaskId id) {
```

Comment:
is this distinction going to be useful to users or us? i know it operates in a separate thread, but so do some other things (e.g. stopping a source connector expects multi-threaded).

Reply:
At least for us I think it does add clarity because these are called by the offset commit thread, not the connector's worker thread. Sometimes I've wanted to just find the messages that were written by all of the code running on the offset commit thread for a particular source connector, and it's not easy to do.

I'm less convinced this will be really obvious to users who are less familiar with the behaviors of Connect. Users do often see log messages related to offset commits and they're not sure whether that affects their source connector and the producer. However, I don't think a separate scope for offsets is detrimental.